### PR TITLE
Fix typos discovered by codespell

### PR DESCRIPTION
**Pull request by cclauss**
[`codespell --ignore-words-list="ans,claus,fith,nam,nd,serie,te" -w .`](https://pypi.org/project/codespell)

### CODE REVIEW

File: chempy/_util.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ def get_backend(backend):
 
 
 def intdiv(p, q):
-    """Integer divsions which rounds toward zero
+    """Integer divisions which rounds toward zero
```
Comment (bjodah) [inline chempy/_util.py:60]: should just read, "division", right?

**Review by bjodah — COMMENTED** [review]
Yikes, thank's for running the spell check. I need to look into integrating some checker with my tools.

File: chempy/util/arithmeticdict.py [hunk 2]
```diff
@@ -23,11 +23,11 @@ def _itruediv(d1, d2):
 
 
 class ArithmeticDict(defaultdict):
-    """A dictionary which supports arithmetics
+    """A dictionary which supports arithmetic
```
Comment (cclauss) [inline chempy/util/arithmeticdict.py:26]: Thoughts on this one?  `codespell` is saying that there is no plural form of this word but we can add it to the `--ignore-words-list` if you want to keep it as is.

File: chempy/_util.py (hunk 1 above)
Comment (cclauss) [inline chempy/_util.py:60]:
```suggestion
    """Integer division which rounds toward zero
```

File: .github/workflows/lint_python.yml [hunk 3]
```diff
@@ -13,7 +13,7 @@ jobs:
                          isort mypy pytest pyupgrade safety
       - run: bandit --recursive --skip B101,B102,B110,B112,B307,B404,B603,B607 .
       - run: black --check . || true
-      - run: codespell --ignore-words-list="ans,nam,nd,serie,te" || true  # --skip="*.css,*.js,*.lock"
+      - run: codespell --ignore-words-list="ans,claus,fith,nam,nd,ond,serie,te"
```
Comment (cclauss) [inline .github/workflows/lint_python.yml:16]: > I need to look into integrating some checker with my tools.

Done.  ;-)

File: chempy/util/arithmeticdict.py (hunk 2 above)
Comment (bjodah) [reply chempy/util/arithmeticdict.py:26]: I think your correction is right (I'm no native speaker, but `arithmetic` sounds better to me, and google agrees)

**Comment by bjodah** [comment]
Great work. Thanks.

**Pull request merged**